lib/vaahextendflutter/services/storage/storage.dart

```diff
@@ -0,0 +1,132 @@
+import '../../env.dart';
+import 'local/flutter_secure_storage/flutter_secure_storage_impl.dart';
+import 'local/hive/hive_storage_impl.dart';
+
+abstract class Storage {
+  static final EnvironmentConfig _envConfig = EnvironmentConfig.getEnvConfig();
+
+  ///Creates a new Local Storage [HiveStorageImpl] or [FlutterSecureStorageImpl]
+  ///
+  ///The argument [name] is used to open Hive box with that name.
+  ///
+  ///If you don't provide [name], 'default' will be used.
+  factory Storage.createLocal({String name = 'default'}) {
+    switch (_envConfig.localStorageType) {
+      case LocalStorageType.hive:
+        final hive = HiveStorageImpl(name: name);
+        hive.init();
```

Comment:
no await here

Reply:
The hive is initialized by the user wherever the `Storage` is used therefore I will remove it.
```diff
- hive.init();
```